# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/ia2/ia2AccessibleText.cpp
# Commit: 5e92b9c7d479
# Full Hash: 5e92b9c7d4794af1d7ab0fb5c278c2e681a18fc6
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2023-01-28 21:11:06
# Regressor Bug: 1811092
# File Overlap Count: 1
# Description:
#   Bug 1811092 - P1: Add new async add/set text selection message. r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D167201
# ==============================================================================

diff -r 7cebaa85f4ff -r 5e92b9c7d479 accessible/windows/ia2/ia2AccessibleText.cpp
--- a/accessible/windows/ia2/ia2AccessibleText.cpp	Fri Jan 27 20:03:06 2023 +0000
+++ b/accessible/windows/ia2/ia2AccessibleText.cpp	Fri Jan 27 20:15:11 2023 +0000
@@ -46,9 +46,9 @@
 
 STDMETHODIMP
 ia2AccessibleText::addSelection(long aStartOffset, long aEndOffset) {
-  auto [textAcc, hr] = LocalTextAcc();
+  HyperTextAccessibleBase* textAcc = TextAcc();
   if (!textAcc) {
-    return hr;
+    return CO_E_OBJNOTCONNECTED;
   }
 
   return textAcc->AddToSelection(aStartOffset, aEndOffset) ? S_OK
@@ -354,9 +354,9 @@
 STDMETHODIMP
 ia2AccessibleText::setSelection(long aSelectionIndex, long aStartOffset,
                                 long aEndOffset) {
-  auto [textAcc, hr] = LocalTextAcc();
+  HyperTextAccessibleBase* textAcc = TextAcc();
   if (!textAcc) {
-    return hr;
+    return CO_E_OBJNOTCONNECTED;
   }
 
   return textAcc->SetSelectionBoundsAt(aSelectionIndex, aStartOffset,